spmd/tensor/parallel/api.py

```diff
@@ -89,6 +94,114 @@ def _shard_self_attn_params(name: str, module: nn.Module) -> None:
                 module.register_module(n, tp_multi_head_attention)
 
 
+# Define partition functions needed to parallelize Linear modules
+def _linear_module_parallelize_row_wise(
+    name: str, module: nn.Linear, device_mesh: DeviceMesh
+) -> None:
+    """
+    This function parallelizes the input :class:``nn.Linear`` module in :class:``RowwiseParallel`` style.
+
+    Args:
+        name (str): name of the input module.
+        module (nn.Module): the :class:``nn.Linear`` object to be parallelized.
+        device_mesh (DeviceMesh): :class:``DeviceMesh`` object which describes the mesh topology
+            of devices for the DTensor.
+    Returns:
+        None
+    """
+    for name, param in module.named_parameters():
+        dist_spec = (
+            [Shard(1)] if name == "weight" else [Replicate()]  # type: ignore[list-item]
+        )
+        dist_param = torch.nn.Parameter(
+            distribute_tensor(param, device_mesh, dist_spec)
+        )
+        module.register_parameter(name, dist_param)
+
+
+def _linear_module_parallelize_col_wise(
+    name: str, module: nn.Linear, device_mesh: DeviceMesh
+) -> None:
+    """
+    This function parallelizes the input :class:``nn.Linear`` module in :class:``ColwiseParallel`` style.
+
+    Args:
+        name (str): name of the input module.
+        module (nn.Module): the :class:``nn.Linear`` object to be parallelized.
+        device_mesh (DeviceMesh): :class:``DeviceMesh`` object which describes the mesh topology
+            of devices for the DTensor.
+    Returns:
+        None
+    """
+    for name, param in module.named_parameters():
+        dist_param = torch.nn.Parameter(
+            distribute_tensor(param, device_mesh, [Shard(0)])
+        )
+        module.register_parameter(name, dist_param)
+
+
+def _parallelize_linear(
+    module: nn.Module,
+    device_mesh: DeviceMesh,
+    parallel_style: ParallelStyle = ColwiseParallel(),
+    tp_mesh_dim: int = 0,
+) -> None:
+    """
+    This function requires that the input module be an object of :class:``nn.Linear``.
+    The module will be parallelized over the n-d :class:``DeviceMesh``
+    based on the :class:``ParallelStyle``.
```

Comment:
I agree that we change all of them to single quote. I can do it in another PR replacing them. what do you think @fduwjj

Reply:
Indeed!! Interesting, because I was looking at the example here: https://www.sphinx-doc.org/en/master/usage/extensions/example_google.html. I thought ``(double is for class) and ` is for variable. Is there any detailed rules for the Google DocString so that we can use as a reference?

Reply:
@fduwjj https://github.com/pytorch/pytorch/blob/master/CONTRIBUTING.md#docstring-type-formatting PyTorch actually uses google style format, so I guess with `:class:` single quote, with no `:class:`, double quote class, single quote variable?

Reply:
Just FYI, 
For requirement "Each line inside a docstrings block must be limited to 80 characters", we can use `black --line-length 80 --preview` to check it.